src/analyze/analyze.c

```diff
@@ -1260,16 +1419,26 @@ static int dot(int argc, char *argv[], void *userdata) {
 
         printf("}\n");
 
-        log_info("   Color legend: black     = Requires\n"
-                 "                 dark blue = Requisite\n"
-                 "                 dark grey = Wants\n"
-                 "                 red       = Conflicts\n"
-                 "                 green     = After\n");
+
+        log_info("   Color legend:\n");
+        for (UnitDependency dependency = UNIT_REQUIRES; dependency < _UNIT_DEPENDENCY_MAX; ++dependency) {
+                if (arg_dependency_color[dependency] && arg_dependency_style[dependency]) {
```

Comment:
hmm, this is a bit sloppy no, if people define a style but no color, no? should deal with one being set and the other omitted

Reply:
There must always be color AND style. The defaults define both of them and the conf files in the analyze.conf.d directories and the cmdline option --dependency-color-style can override some or all of them.

Reply:
Now it's possible to define colors independently from style, options --dependency-color= and --dependency-style=